### PR TITLE
Reorganize sync media page and add alert when switching sync mode

### PR DESCRIPTION
**Pull request by dugajean**
Screen Shot 2020-07-29 at 15 12 08

### CODE REVIEW

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/css/src/components/_settings.scss [hunk 1]
```diff
@@ -216,4 +216,23 @@
 	border: 1px solid #ccd0d4;
 	border-left: 4px solid #00a0d2;
 	box-shadow: 0 1px 1px rgba(0,0,0,.04);
+}
+
+.field-radio {
+	input[type=radio] {
+		margin: 0;
+		margin-right: 5px !important;
```
Comment (pereirinha) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/css/src/components/_settings.scss:224]: @dugajean 
Can we not use the `!important`? I suspect that in this case, you'll need to increase the declaration specificity — I tested quickly adding the class `cld-field` and it was enough.

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/css/src/components/_settings.scss [hunk 2]
```diff
@@ -216,4 +216,23 @@
...
+	}
+
+	label {
+		margin-right: 10px;
+	}
+}
+
+.settings-tab-section h2 {
+	margin: 0 !important;
```
Comment (pereirinha) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/css/src/components/_settings.scss:233]: Same here, though in this case, I didn't notice any difference without the `!important`.

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/components/settings-page.js [hunk 3]
```diff
@@ -79,7 +79,13 @@
 					field.removeClass( 'settings-valid-field' );
 					field.addClass( 'settings-invalid-field' );
 				}
-			}).trigger('change' );
+			}).trigger('change');
+
+			$('[name="cloudinary_sync_media[auto_sync]"]').change(function () {
+				if ($(this).val() === 'on') {
+					alert(cloudinaryJsData.auto_sync_notice)
```
Comment (pereirinha) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/components/settings-page.js:86]: @dugajean 

Alerts works just fine, but WordPress has a modal system, the [Thickbox](https://codex.wordpress.org/Javascript_Reference/ThickBox) — that brings a more seamless experience. What do you think?

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-settings-page.php [hunk 4]
```diff
@@ -161,6 +161,11 @@ private function register_section( $tab_slug ) {
 		register_setting( $setting_slug, $setting_slug, $args );
 		add_filter( 'pre_update_site_option_' . $setting_slug, array( $this, 'set_notices' ), 10, 3 );
 		add_filter( 'pre_update_option_' . $setting_slug, array( $this, 'set_notices' ), 10, 3 );
+
+		if ( ! empty( $tab['javascript_i18n'] ) ) {
+			wp_localize_script( 'jquery', 'cloudinaryJsData', $tab['javascript_i18n'] );
```
Comment (pereirinha) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-settings-page.php:166]: @dugajean 

Shouldn't we use the dependency on the plugin script instead `cloudinary-settings`?

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/css/src/components/_settings.scss (hunk 1 above)
Comment (dugajean) [reply cloudinary-image-management-and-manipulation-in-the-cloud-cdn/css/src/components/_settings.scss:224]: Thanks, will change it!

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/components/settings-page.js (hunk 3 above)
Comment (dugajean) [reply cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/components/settings-page.js:86]: Oh I wasn't aware. I'll use this 😉

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-settings-page.php [hunk 5]
```diff
@@ -290,6 +296,14 @@ public function render_field( $field, $value = null, $show_description = true )
...
 				<?php
 				break;
+			case 'radio':
+				foreach ( $field['choices'] as $key => $option ) :
+				?>
+					<input type="<?php echo esc_attr( $type ); ?>" class="cld-field regular-<?php echo esc_attr( $type ); ?>" id="<?php echo esc_attr( $field['label_for'] . '_' . $key ); ?>" name="<?php echo esc_attr( $setting_slug ); ?>[<?php echo esc_attr( $field['slug'] ); ?>]" <?php if ( ! empty( $field['pattern'] ) ) : ?>pattern="<?php echo esc_attr( $field['pattern'] ); ?>"<?php endif; ?> data-condition="<?php echo esc_attr( $condition ); ?>" data-context="<?php echo esc_attr( $context ); ?>" <?php echo esc_attr( $required ); ?> <?php checked( $key, $value ); ?> value="<?php echo esc_attr( $key ); ?>"> 
+					<label for="<?php echo esc_attr( $field['label_for'] . '_' . $key ); ?>"><?php echo $option; ?></label>
```
Comment (pereirinha) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-settings-page.php:303]: @dugajean 

It seems that we aren't escaping `$option`.

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-settings-page.php (hunk 4 above)
Comment (pereirinha) [reply cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-settings-page.php:166]: Do we still need this object?

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-settings-page.php (hunk 5 above)
Comment (dugajean) [reply cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-settings-page.php:303]: Good catch 👍

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-settings-page.php [hunk 6]
```diff
@@ -290,6 +296,14 @@ public function render_field( $field, $value = null, $show_description = true )
...
 				<?php
 				break;
+			case 'radio':
+				foreach ( $field['choices'] as $key => $option ) :
+				?>
+					<input type="<?php echo esc_attr( $type ); ?>" class="cld-field regular-<?php echo esc_attr( $type ); ?>" id="<?php echo esc_attr( $field['label_for'] . '_' . $key ); ?>" name="<?php echo esc_attr( $setting_slug ); ?>[<?php echo esc_attr( $field['slug'] ); ?>]" <?php if ( ! empty( $field['pattern'] ) ) : ?>pattern="<?php echo esc_attr( $field['pattern'] ); ?>"<?php endif; ?> data-condition="<?php echo esc_attr( $condition ); ?>" data-context="<?php echo esc_attr( $context ); ?>" <?php echo esc_attr( $required ); ?> <?php checked( $key, $value ); ?> value="<?php echo esc_attr( $key ); ?>"> 
+					<label for="<?php echo esc_attr( $field['label_for'] . '_' . $key ); ?>"><?php esc_html( $option ); ?></label>
```
Comment (DavidCramer) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-settings-page.php:303]:
```suggestion
					<label for="<?php echo esc_attr( $field['label_for'] . '_' . $key ); ?>"><?php esc_html_e( $option ); ?></label>
```

**Pull request merged**